### PR TITLE
Fix products visibility

### PR DESCRIPTION
**Pull request by IKarbowiak**
Do not require `MANAGE_PRODUCTS` for almost everything. It was impossible to open the draft order, sale, or voucher edit view without `MANAGE_PRODUCTS` permission. Right now it requires a staff user.

Linked tasks: [SALEOR-1693](https://app.clickup.com/t/2549495/SALEOR-1693), [SALEOR-1998](https://app.clickup.com/t/2549495/SALEOR-1998)

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1693 Dashboard user can't add unlisted products to orders](https://app.clickup.com/t/2549495/SALEOR-1693)

**Comment by patrys** [comment]
Task linked: [SALEOR-1998 Dashboard use `products` query without "MANAGE_PRODUCTS" and "channel" in args.](https://app.clickup.com/t/2549495/SALEOR-1998)

**Review by d-wysocki — APPROVED** [review]

**Review by tomaszszymanski129 — APPROVED** [review]

File: saleor/graphql/menu/types.py [hunk 1]
```diff
@@ -104,10 +104,8 @@ def resolve_collection(root: ChannelContext[models.MenuItem], info, **_kwargs):
             return None
 
         requestor = get_user_or_app_from_context(info.context)
-        requestor_has_access_to_all = Collection.objects.user_has_access_to_all(
-            requestor
-        )
-        if requestor_has_access_to_all:
+        staff_member = requestor_is_staff_member(requestor)
```
Comment (maarcingebala) [inline saleor/graphql/menu/types.py:107]: Maybe rename `staff_member` to `is_staff_member` since it's a boolean?

File: saleor/account/utils.py [hunk 2]
```diff
@@ -77,3 +82,9 @@ def remove_staff_member(staff):
         staff.save()
     else:
         staff.delete()
+
+
+def requestor_is_staff_member(requestor: Union[User, "App"]):
+    """Return true if requestor is an active app or active staff user."""
+    is_staff = getattr(requestor, "is_staff", True)
```
Comment (maarcingebala) [inline saleor/account/utils.py:89]: Why default `True` here? It would be safer to assume `False` when we are not sure the field `is_staff` is present.

Comment (IKarbowiak) [reply saleor/account/utils.py:89]: I assumed that, as `App` does not have `is_staff` field, so the `App` instance is always treated as a staff member. But maybe check with `isinstance` will be better.

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**